cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/StrimziPodSetController.java

```diff
@@ -0,0 +1,491 @@
+                podCounter.readyPods++;
+            }
+
+            if (!PodRevision.hasChanged(currentPod, pod))    {
+                podCounter.currentPods++;
+            }
+
+            // TODO: Add patching of exiting pods => to be done in the future to handle selected changes to the Pods
+            //  which might not require rolling updates
+        }
+    }
+
+    /**
+     * Removes the pods which were removed from the StrimziPodSet but which match the selector.
+     *
+     * @param reconciliation    Reconciliation in which this is executed
+     * @param selector          LabelSelector to match the pods belonging to this StrimziPodSet
+     * @param desiredPodNames   Collection with names of the pods which are still desired
+     * @param podCounter        Pod Counter used to count pods for the status
+     */
+    private void removeDeletedPods(Reconciliation reconciliation, LabelSelector selector, Collection<String> desiredPodNames, PodCounter podCounter) {
+        Set<String> toBeDeleted = podLister
+                .namespace(reconciliation.namespace())
+                .list()
+                .stream()
+                .filter(pod -> Util.matchesSelector(Optional.of(selector), pod))
+                .map(pod -> pod.getMetadata().getName())
+                .collect(Collectors.toSet());
+        toBeDeleted.removeAll(desiredPodNames);
+
+        for (String podName : toBeDeleted)  {
+            LOGGER.debugCr(reconciliation, "Deleting pod {} in namespace {}", podName, reconciliation.namespace());
+            podOperator.client().inNamespace(reconciliation.namespace()).withName(podName).withPropagationPolicy(DeletionPropagation.BACKGROUND).delete();
+            podCounter.pods++;
+        }
+    }
+
+    /**
+     * Enqueues the next reconciliation. It checks whether another reconciliation for the same resource is already in
+     * the queue and enqueues the new event only if it is not there yet.
+     *
+     * @param reconciliation    Reconciliation identifier
+     */
+    private void enqueue(SimplifiedReconciliation reconciliation)    {
+        if (!workQueue.contains(reconciliation)) {
+            LOGGER.debugOp("Enqueueing StrimziPodSet {} in namespace {}", reconciliation.name, reconciliation.name);
+            workQueue.add(reconciliation);
+        } else {
+            LOGGER.debugOp("StrimziPodSet {} in namespace {} is already enqueued => ignoring", reconciliation.name, reconciliation.name);
+        }
+    }
+
+    /**
+     * The run loop of the controller thread. It picks reconciliations from the work queue and executes them.
+     */
+    @Override
+    public void run() {
+        LOGGER.infoOp("Starting StrimziPodSet controller for namespace {}", watchedNamespace);
+
+        LOGGER.infoOp("Waiting for informers to sync");
+        while (!stop
+                && !podInformer.hasSynced()
+                && !strimziPodSetInformer.hasSynced()
+                && kafkaInformer.hasSynced())   {
```

Comment:
should you wait for `kafkaInformer` to be synced? so the condition `!kafkaInformer.hasSynced()` ?